### PR TITLE
Adding Property Definitions section to the lefout

### PR DESCRIPTION
**Pull request by osamahammad21**

### CODE REVIEW

File: src/lefout/lefout.cpp [hunk 1]
```diff
@@ -811,6 +814,172 @@ void lefout::writeMTerm(dbMTerm* mterm)
...
+    break;
+  default:
+    return;
+  }
+  dbSet<dbProperty> properties = dbProperty::getProperties(obj); 
+  dbSet<dbProperty>::iterator pitr;
+  for (pitr = properties.begin(); pitr != properties.end(); ++pitr)
+  {
+    dbProperty* prop = *pitr;
+    if(propertiesMap.find(prop->getName())==propertiesMap.end())
```
Comment (maliberty) [inline src/lefout/lefout.cpp:909]: operator[] should default initialize the value so I don't think this is necessary.

File: src/lefout/lefout.cpp [hunk 2]
```diff
@@ -811,6 +814,172 @@ void lefout::writeMTerm(dbMTerm* mterm)
...
+    writePropertyDefinition(prop);      
+  }
+}
+
+void lefout::writePropertyDefinitions(dbLib* lib)
+{
+  std::unordered_map<std::string,short> propertiesMap;
+  dbTech* tech = lib->getDb()->getTech();
+  fprintf(_out, "\nPROPERTYDEFINITIONS\n");
+  //Layer is bit 0 in the short value of the map
```
Comment (maliberty) [inline src/lefout/lefout.cpp:923]: Why is the bit number relevant here?  That seems like an implementation detail of writeObjectPropertyDefinitions

File: src/lefout/lefout.cpp [hunk 3]
```diff
@@ -811,6 +814,172 @@ void lefout::writeMTerm(dbMTerm* mterm)
...
+  dbSet<dbTechLayer> layers = tech->getLayers();
+  dbSet<dbTechLayer>::iterator litr;
+  for (litr = layers.begin(); litr != layers.end(); ++litr)
```
Comment (maliberty) [inline src/lefout/lefout.cpp:926]: Use a range-for style instead:

for (dbTechLayer* layer : tech->getLayers()) { ... }

**Review by maliberty — CHANGES_REQUESTED** [review]

File: src/lefout/lefout.cpp (hunk 3 above)
Comment (osamahammad21) [reply src/lefout/lefout.cpp]: Done

File: src/lefout/lefout.cpp (hunk 2 above)
Comment (osamahammad21) [reply src/lefout/lefout.cpp]: Done

File: src/lefout/lefout.cpp (hunk 1 above)
Comment (osamahammad21) [reply src/lefout/lefout.cpp]: Done

**Review by maliberty — CHANGES_REQUESTED** [review]
One last thing : please run clang-format on the code to match it match stylistically.

**Review by maliberty — APPROVED** [review]

**Pull request merged**